### DISCUSSION TITLE
Some help to learn do use DSL and map class

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#433] [Q&A] [answered]
I have this input

```
field_a=2,3,field_b=a,b,c#a,b
```

My first goal is to split first field by `,`, and I will have `field_a[0]=2` and `field_a[1]=3` and to split `field_b` by `#` and I will have `field_b[0]="a,b,c"` and `field_b[1]="a,b"`.
The second one is to couple the two fields by index (`0` and `1`) to have at the end

```
field_a=2,field_b=a,b,c
field_a=3,field_b=a,b
```

I imagine I should use some map class function as `splitnvx` coupled with some for loop. But I am not able to even take a first small step.

Thank you

**Comment by johnkerl** [comment, marked as answer]
I think this is going to be hard since in `field_a=2,3,field_b=a,b,c#a,b` the commas are playing the double role of separating `field_a` stuff from `field_b` stuff, and, separating `2` from `3` and likewise `a`, `b`, `c#a`, `b`.

If I understand correctly the data structure is of the form

```
{
  "field_a" : [2, 3],
  "field_b": ["a", "b", "c#a", "b"]
}
```

and I don't know how to process this data that way, unless either you use a file format that supports nesting (like JSON), or, change one of the delimiters -- for example with a semicolon:

```
field_a=2,3;field_b=a,b,c#a,b
```

**Reply by aborruso** [reply]
Thank you.

I have made a big mistake creating the sample file. I have used wrong separator and format.

If my input was `{ "field_a": "2,3", "field_b": "a_b_c,a_b" }`, would there be a way to transform it in 

```
{ "field_a": 2, "field_b": "a_b_c" }
{ "field_a": 3, "field_b": "a_b" }
```

**Reply by johnkerl** [reply]
Hmm, this is a lot like `nest --explode --values --across-records` but not quite.

How's this?

Input data:

```
$ cat xx
field_a:1,2;field_b:s,t
field_a:3,4;field_b:u,v
```

Echo back input data:

```
$ mlr --ips colon --ifs semicolon --opprint cat xx
field_a field_b
1,2     s,t
3,4     u,v
```

Split:

```
../c/mlr --ips colon --ifs semicolon --opprint --from xx put -q '
  asplits = splitnv($field_a, ",");
  bsplits = splitnv($field_b, ",");
  n = length(asplits);
  for (int i = 1; i <= n; i += 1) {
    outrec = $*;
    outrec["field_a"] = asplits[i];
    outrec["field_b"] = bsplits[i];
    emit outrec;
  }
'

field_a field_b
1       s
2       t
3       u
4       v
```

**Reply by aborruso** [reply]
Thank you very mach, very useful and simple. But I'm a goat, I'm still at a very low level with DLS.

SOLVED!!

**Reply by aborruso** [reply]
My secret goal was to reply to this question https://stackoverflow.com/a/66412025/757714
